tikv-client/src/main/java/com/pingcap/tikv/key/IndexKey.java

```diff
@@ -42,54 +43,51 @@ private IndexKey(long tableId, long indexId, Key[] dataKeys) {
   }
 
   public static class EncodeIndexDataResult {
-    public EncodeIndexDataResult(Key[] keys, boolean appendHandle) {
+    public EncodeIndexDataResult(byte[] keys, boolean appendHandle) {
       this.keys = keys;
       this.appendHandle = appendHandle;
     }
 
-    public Key[] keys;
+    public byte[] keys;
     public boolean appendHandle;
   }
 
   public static IndexKey toIndexKey(long tableId, long indexId, Key... dataKeys) {
     return new IndexKey(tableId, indexId, dataKeys);
   }
 
-  public static EncodeIndexDataResult encodeIndexDataValues(
-      Row row,
-      List<TiIndexColumn> indexColumns,
-      Handle handle,
-      boolean appendHandleIfContainsNull,
-      TiTableInfo tableInfo) {
+  public static EncodeIndexDataResult genIndexKey(
+      long physicalID, Row row, TiIndexInfo indexInfo, Handle handle, TiTableInfo tableInfo) {
     // when appendHandleIfContainsNull is true, append handle column if any of the index column is
     // NULL
```

Comment:
your logical has changed, the code annotation needs be changed too

Reply:
> your logical has changed, the code annotation needs be changed too

changed the annotation